File: Pertemuan 3/Mahasiswa.py
# Nama : Salma Adiba Chasani
# NIM  : 32602400123

class Mahasiswa:
    def __init__(self):
        self.__nama = None
        self.__nim = None

    # Getter & Setter Nama
    def get_nama(self):
        return self.__nama

    def set_nama(self, nama):
        self.__nama = nama

    # Getter & Setter NIM dengan validasi
    def get_nim(self):
        return self.__nim

    def set_nim(self, nim):
        if isinstance(nim, str) and nim.isdigit() and len(nim) == 10:
            self.__nim = nim
        else:
            raise ValueError("NIM harus 10 digit angka!")

# === PROGRAM UTAMA ===
m = Mahasiswa()

m.set_nama(input("Masukkan Nama: "))

valid = False
while not valid:
    try:
        nim = input("Masukkan NIM (10 digit): ")
        m.set_nim(nim)
        valid = True
    except ValueError as e:
        print(e)

print("\n=== DATA MAHASISWA ===")
print("Nama:", m.get_nama())
print("NIM :", m.get_nim())

File: Pertemuan 3/Mahasiswa/src/mahasiswa/Mahasiswa.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package mahasiswa;
import java.util.Scanner;
/**
 *
 * @author salma
 */

// Nama : Salma Adiba Chasani 
// NIM  : 32602400123
public class Mahasiswa {

    private String nama;
    private String nim;

    // Getter & Setter Nama
    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    // Getter & Setter NIM dengan validasi
    public String getNim() {
        return nim;
    }

    public void setNim(String nim) {
        if (nim != null && nim.matches("\\d{10}")) {
            this.nim = nim;
        } else {
            throw new IllegalArgumentException("NIM harus 10 digit angka!");
        }
    }

    // untuk mengisi data mahasiswa
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        Mahasiswa m = new Mahasiswa();

        System.out.print("Masukkan Nama: ");
        m.setNama(input.nextLine());

        boolean validNim = false;
        while (!validNim) {
            try {
                System.out.print("Masukkan NIM (10 digit): ");
                m.setNim(input.nextLine());
                validNim = true;
            } catch (IllegalArgumentException e) {
                System.out.println(e.getMessage());
            }
        }

        System.out.println("\n=== DATA MAHASISWA ===");
        System.out.println("Nama: " + m.getNama());
        System.out.println("NIM : " + m.getNim());
    }
    
}
